src/mlpack/tests/distribution_test.cpp

```diff
@@ -390,10 +390,145 @@ BOOST_AUTO_TEST_CASE(GaussianDistributionTrainTest)
       BOOST_REQUIRE_SMALL(d.Covariance()(i, j) - actualCov(i, j), 1e-5);
 }
 
+/**
+  * This test verifies the fitting of GaussianDistribution
+  works properly when probabilities for each sample is given.
+**/
+BOOST_AUTO_TEST_CASE(GaussianDistributionTrainWithProbabilitiesTest)
+{
+  double mean = 5.0;
+  double stddeviation = 2.0;
+
+  //Creates a normal distribution generator.
+  std::default_random_engine generator;
+  generator.seed(std::time(NULL));
+  std::normal_distribution<double> dist(mean, stddeviation);
+  
+  size_t N = 50000;
```

Comment:
I think only 5k samples should be fine for this test---no need to use more samples than are needed.